src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.MethodSymbolKey.cs

```diff
@@ -20,14 +21,19 @@ public static void Create(IMethodSymbol symbol, SymbolKeyWriter visitor)
 
             public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? failureReason)
             {
-                var reducedFromResolution = reader.ReadSymbolKey(out var reducedFromFailureReason);
+                // Methods are members, and members are never resolved in a way where we have contextual types to guide
+                // how the outer parts of the member may resolve.  We can use contextual typing for the *signature*
+                // portion of the member though.
+
+                var reducedFromResolution = reader.ReadSymbolKey(contextualType: null, out var reducedFromFailureReason);
+                var receiverTypeResolution = reader.ReadSymbolKey(contextualType: null, out var receiverTypeFailureReason);
+
                 if (reducedFromFailureReason != null)
                 {
                     failureReason = $"({nameof(ReducedExtensionMethodSymbolKey)} {nameof(reducedFromResolution)} failed -> {reducedFromFailureReason})";
                     return default;
                 }
 
-                var receiverTypeResolution = reader.ReadSymbolKey(out var receiverTypeFailureReason);
```

Comment:
this was incorrect logic before.  symbolkeys must read their keys to the end, and only return failures once they've done this.  otherwise, they end up reading partway through their own key and returning to the caller, who will have no idea how to resume from that point on.

Reply:
No tests changed, so was this only a theoretical bug that can't actually happen in real life, or just poor test coverage? And if the latter...

Reply:
Theoretical bug